xwayland/sockets.c

```diff
@@ -84,11 +84,44 @@ static int open_socket(struct sockaddr_un *addr, size_t path_size) {
 	return -1;
 }
 
+static bool check_socket_dir(void) {
+	struct stat buf;
+
+	if (lstat(socket_dir, &buf)) {
+		wlr_log_errno(WLR_ERROR, "Failed to stat %s", socket_dir);
+		return false;
+	}
+	if (!(buf.st_mode & S_IFDIR)) {
+		wlr_log(WLR_ERROR, "%s is not a directory", socket_dir);
+		return false;
+	}
+	if (!((buf.st_uid == 0) || (buf.st_uid == getuid()))) {
+		wlr_log(WLR_ERROR, "%s not owned by root or us", socket_dir);
+		return false;
+	}
+	if (!(buf.st_mode & S_ISVTX)) {
+		if ((buf.st_mode & (S_IWGRP | S_IWOTH))) {
```

Comment:
Note to self: "Bail out if the sticky bit is not set and the directory is writable by other users"

(Maybe would be nice to turn this into a comment.)